ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/ccd/services/CoreCaseDataService.java

```diff
@@ -0,0 +1,130 @@
+    public static final String CASE_TYPE_ID = "MoneyClaimCase";
+
+    private static final Logger logger = LoggerFactory.getLogger(MigrateCoreCaseDataService.class);
+
+    private final CoreCaseDataApi coreCaseDataApi;
+    private final AuthTokenGenerator authTokenGenerator;
+    private final MigrateCoreCaseDataService migrateCoreCaseDataService;
+
+    @Autowired
+    public CoreCaseDataService(
+        MigrateCoreCaseDataService migrateCoreCaseDataService,
+        CoreCaseDataApi coreCaseDataApi,
+        AuthTokenGenerator authTokenGenerator
+    ) {
+        this.migrateCoreCaseDataService = migrateCoreCaseDataService;
+        this.coreCaseDataApi = coreCaseDataApi;
+        this.authTokenGenerator = authTokenGenerator;
+    }
+
+    public void create(User user, Claim claim) {
+        logger.info("Create case in CCD, claim id = " + claim.getId());
+        try {
+            EventRequestData eventRequestData = EventRequestData.builder()
+                .userId(user.getUserDetails().getId())
+                .jurisdictionId(JURISDICTION_ID)
+                .caseTypeId(CASE_TYPE_ID)
+                .eventId(EventType.MIGRATED_FROM_CLAIMSTORE.getValue())
+                .ignoreWarning(true)
+                .build();
+
+            migrateCoreCaseDataService.save(
+                user.getAuthorisation(),
+                eventRequestData,
+                claim
+            );
+        } catch (Exception exception) {
+            throw new CreateCaseException(
+                String.format("Failed storing claim in CCD store for claim %s", claim.getReferenceNumber()), exception
+            );
+        }
+    }
+
+    public void overwrite(User user, Long ccdId, Claim claim) {
+        logger.info("Overwrite " + ccdId + ", claim id = " + claim.getId());
+        try {
+            EventRequestData eventRequestData = EventRequestData.builder()
+                .userId(user.getUserDetails().getId())
+                .jurisdictionId(JURISDICTION_ID)
+                .caseTypeId(CASE_TYPE_ID)
+                .eventId(EventType.MIGRATED_FROM_CLAIMSTORE.getValue())
+                .ignoreWarning(true)
+                .build();
+
+            migrateCoreCaseDataService.update(user.getAuthorisation(), eventRequestData, ccdId, claim);
+        } catch (Exception exception) {
+            throw new OverwriteCaseException(
+                String.format(
+                    "Failed updating claim in CCD store for claim %s on event %s",
+                    claim.getReferenceNumber(),
+                    EventType.MIGRATED_FROM_CLAIMSTORE), exception
+            );
+        }
+    }
+
+    public Optional<Long> getCcdIdByReferenceNumber(User user, String referenceNumber) {
+        logger.info("Get claim from CCD " + referenceNumber);
+
+        Optional<Long> ccdId = search(user, ImmutableMap.of("case.referenceNumber", referenceNumber));
+        ccdId.ifPresent((id) -> logger.info("Claim found " + id));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "id" parameter [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AS1611)